### PR TITLE
[Debugger] Add variable querying related test

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will add variable querying related tests for the following variables:

- [x] arrays
- [x] JSON
- [x] maps
- [x] XML
- [x] tables
- [x] tuples

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/28451

### CODE REVIEW

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java [hunk 1]
```diff
@@ -542,15 +542,23 @@ public boolean waitForDebugTermination(long timeoutMillis) {
      * Can be used to get child variables from parent variable.
      *
      * @param parentVariable parent variable
+     * @param start          starting index of child variable
+     * @param count          number of child variable count
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java:546]: Not formatted properly

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java [hunk 2]
```diff
@@ -560,6 +568,19 @@ public boolean waitForDebugTermination(long timeoutMillis) {
...
+     * Can be used to get child variables from parent variable.
+     *
+     * @param parentVariable parent variable
+     * @return variable map with child variables information
+     * @throws BallerinaTestException if an error occurs when fetching debug hit child variables
+     */
+    public Map<String, Variable> fetchChildVariables(Variable parentVariable) throws BallerinaTestException {
+        int start = -1;
+        int count = -1;
+        return this.fetchChildVariables(parentVariable, start, count);
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java:581]:
```suggestion
        return fetchChildVariables(parentVariable, -1, -1);
```

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Test class for variable queries.
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java:38]:
```suggestion
 * Test class for variable paging scenarios.
```

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    public void xmlVariableQueryTest() throws BallerinaTestException {
+        // xml child variable query test
+        Map<String, Variable> xmlChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("xmlVar"));
+        Assert.assertEquals(xmlChildVariables.size(), 1000);
+
+        // xml child variable query test with start = 0 and count = 0
+        xmlChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("xmlVar"), 0, 0);
+        Assert.assertEquals(xmlChildVariables.size(), 1000);
+
+    }
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java:186]: Don't we need to check intermediate ranges for xml, just like we have done for other types?

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    @AfterClass(alwaysRun = true)
+    public void cleanUp() {
+        debugTestRunner.terminateDebugSession();
+    }
+
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java:192]: Additional blank line

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    @Test
+    public void xmlVariableQueryTest() throws BallerinaTestException {
+        // xml child variable query test
+        Map<String, Variable> xmlChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("xmlVar"));
+        Assert.assertEquals(xmlChildVariables.size(), 1000);
+
+        // xml child variable query test with start = 0 and count = 0
+        xmlChildVariables = debugTestRunner.fetchChildVariables(localVariables.get("xmlVar"), 0, 0);
+        Assert.assertEquals(xmlChildVariables.size(), 1000);
+
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java:185]: additional blank line

File: tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java (hunk 2 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/main/java/org/ballerinalang/debugger/test/utils/DebugTestRunner.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/32015/commits/b385f1235a8159e68de0c9353d344ad184ccd6a4

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java (hunk 3 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/32015/commits/b385f1235a8159e68de0c9353d344ad184ccd6a4

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java (hunk 4 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/32015/commits/b385f1235a8159e68de0c9353d344ad184ccd6a4

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java (hunk 5 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/32015/commits/b385f1235a8159e68de0c9353d344ad184ccd6a4

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java (hunk 6 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/variables/VariableQueryTest.java:193]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/32015/commits/b385f1235a8159e68de0c9353d344ad184ccd6a4

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**